### PR TITLE
[ROCM] Updated fp8 matmul with adjustments for updated hipBlasLt support in ROCm 6.2

### PR DESCRIPTION
**Pull request by wenchenvincent**
Main changes include:
* Added support for fp8 matmul with output data type to be fp8 and bf16.
* Added buffer comparators for fp8e4m3fnuz and fp8e5m2fnuz

### CODE REVIEW

**Comment by wenchenvincent** [comment]
@reedwm Could you kindly take a look at this PR? Thanks!

**Review by reedwm — CHANGES_REQUESTED** [review]

File: xla/service/gpu/gemm_rewriter.cc [hunk 1]
```diff
@@ -1098,16 +1099,26 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
     switch (instr->shape().element_type()) {
       case F8E4M3FN:
```
[the remark this answers is not in the document]
Comment (wenchenvincent) [reply xla/service/gpu/gemm_rewriter.cc:1100]: I agree. Before adding that, I have a question: `gemm_rewriter.cc` was refactored to get rid of those `GOOGLE_CUDA` and `TENSORFLOW_USE_ROCM` compile-time macros and use runtime checks of `gpu_version_` and `toolkit_version_`. I followed the new style mostly except here, as using the runtime checks of `gpu_version` and `toolkit_version_` might make the code a bit more duplicated here. What's your opinion about this?

[the remark this answers is not in the document]
Comment (reedwm) [reply xla/service/gpu/gemm_rewriter.cc:1100]: Ah I didn't realize GOOGLE_CUDA and TENSORFLOW_USE_ROCM are no longer used. I would continue to not use them. But yeah, this does make writing the switch statement a bit tricky.

I would replace the switch statement with an if-statement. Do something like:

```c++
PrimitiveType d_type = instr->shape().element_type();
bool supported_d_type = d_type == BF16 || d_type == F16 || d_type == F32;
if ((IsCuda() && (d_type == F8E4M3FN || d_type == F8E5M2)) {
  supported_d_dtype = true
}
if ((IsRocm() && rocm_version >= 60200 && (d_type == F8E4M3FNUZ || d_type == F8E5M2FNUZ)) {
  supported_dtype = true
}
If (!supported_dtype) {
  // VLOG and return false
}
```

Alternatively, the switch-statement can be kept by using fallthrough: Only break in the F8E4M3FNUZ and F8E5M2FNUZ case if a supported rocm version is used.

[the remark this answers is not in the document]
Comment (wenchenvincent) [reply xla/service/gpu/gemm_rewriter.cc]: Thanks! I think the if-statement is clearer and just changed to it.

**Review by reedwm — APPROVED** [review]

**Pull request closed without merging**